ksqldb-functional-tests/src/test/resources/query-validation-tests/self-joins.json

```diff
@@ -0,0 +1,513 @@
+{
+  "tests": [
+    {
+      "name": "self join - two streams - with both join column in projection",
+      "statements": [
+        "CREATE STREAM L (A INT KEY, B INT, C INT) WITH (kafka_topic='LEFT', value_format='JSON');",
+        "CREATE STREAM R (A INT KEY, B INT, C INT) WITH (kafka_topic='LEFT', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT * FROM L INNER JOIN R WITHIN 10 SECONDS ON L.A = R.A;"
```

Comment:
Can you add the `GRACE PERIOD` on all the tests? Joins with no grace period are deprecated and will be removed in the future.

```
ksql> CREATE STREAM OUTPUT AS SELECT * FROM L INNER JOIN R WITHIN 10 SECONDS ON L.A = R.A;

 Message                              
--------------------------------------
 Created query with ID CSAS_OUTPUT_11 
--------------------------------------
WARNING: DEPRECATION NOTICE: Stream-stream joins statements without a GRACE PERIOD will not be accepted in a future ksqlDB version.
Please use the GRACE PERIOD clause as specified in https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/select-push-query/
```